src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.spec.ts

```diff
@@ -107,35 +107,28 @@ describe('OsdListComponent', () => {
   });
 
   describe('test table actions in submenu', () => {
+    let tableActionDebugElement;
+    let toClassName;
+
+    const getActionClasses = (action) =>
+      tableActionDebugElement.query(By.css('.' + toClassName(action.name))).classes;
+
     beforeEach(
       fakeAsync(() => {
         // The menu needs a click to render the dropdown!
         const dropDownToggle = fixture.debugElement.query(By.css('.dropdown-toggle'));
         dropDownToggle.triggerEventHandler('click', null);
         tick();
         fixture.detectChanges();
+        tableActionDebugElement = fixture.debugElement.query(By.directive(TableActionsComponent));
+        toClassName = TestBed.get(TableActionsComponent).toClassName;
       })
     );
 
-    /**
-     * Helper function to retrieve menu item
-     * @param selector
-     */
-    const getMenuItem = (selector: string): DebugElement => {
-      return fixture.debugElement
-        .query(By.directive(TableActionsComponent))
-        .query(By.css(selector));
-    };
-
-    it('has menu entries disabled for entries without create permission', () => {
-      component.tableActions
-        .filter((tableAction) => tableAction.permission !== 'create')
-        .map((tableAction) => tableAction.name)
-        .map(TestBed.get(TableActionsComponent).toClassName)
-        .map((className) => getMenuItem(`.${className}`))
-        .forEach((debugElement) => {
-          expect(debugElement.classes.disabled).toBe(true);
-        });
+    it('has all menu entries disabled', () => {
```

Comment:
The new code removes a feature of the old one: It doesn't take into account that actions with `create` permissions aren't supposed to be disabled by default. Can you please re-add it?

Reply:
That's right, it has been removed and will be implemented with a later PR. I know. We can re-add it later, but due to the structure of the dashboard, especially the TableActionComponent and the way the permissions are grouped to create, read, update and write, it is very unlikely that this code has to be touched if any "action" with create permission will be added. IMO it makes perfectly sense to keep it and that it doesn't need to be touched.

But if that doesn't convice you and your desire is a temporary removal of the filter, I'd like to suggest you to simply remove the line that filters the code based on their permissions and change the description of the `it` function accordingly.

Remove
```typescript
        .filter((tableAction) => tableAction.permission !== 'create')
```
adapt the description and it would look like
```typescript
    it('consists of disabled menu entries', () => {
      component.tableActions
        .map((tableAction) => tableAction.name)
        .map(TestBed.get(TableActionsComponent).toClassName)
        .map((className) => getMenuItem(`.${className}`))
        .forEach((debugElement) => {
          expect(debugElement.classes.disabled).toBe(true);
        });
    });

```

Reply:
If you add a create action simply change `.toBe(true)` to `.toBe(action.permission === 'create' ? undefined : true)`.

Reply:
I agree with @Devp00l that the original description of the test case is misleading, but I'm fine with @p-na's suggested patch.

Reply:
New code seems a bit simpler.
I'm OK with keeping the `filter`, but we can also just add it later.
No need to change the expect condition.

Reply:
My suggestion to change the expected condition is that all items will be tested so that the `create` action (which not exists yet) will be tested too. If we would not include it here and it would be disabled which it shouldn't be, we would never now if we filter those actions out.